zenpy/lib/generator.py

```diff
@@ -47,7 +49,10 @@ def next(self):
             self.handle_pagination()
         if len(self.values) < 1:
             raise StopIteration()
+        if self.count is not None and self._iteration > self.count:
+            raise StopIteration()
```

Comment:
I don't disagree with this as a halting condition as long as this count doesn't change :)

Reply:
cc [dolbinskaBW](https://github.com/dolbinskaBW)

Reply:
I think I probably need to rework this, I was able to get some time to try a large extract from Talk and the count is per page, so if there was 1234 rows it was showing as 1000, then 234 on the next page. I was unable to use Zenpy directly because of this issue but yeah I don't think my fix is ready.

Apologies been swamped the last month but hopefully I get a bit to wrap this up.